orchestration/hca_orchestration/pipelines/set_snapshot_public.py

```diff
@@ -0,0 +1,112 @@
+            "gcs": google_storage_client,
+            "hca_manage_config": preconfigure_resource_for_mode(hca_manage_config, hca_env),
+            "io_manager": preconfigure_resource_for_mode(gcs_pickle_io_manager, hca_env),
+            "sam_client": preconfigure_resource_for_mode(sam_client, hca_env),
+            "slack": preconfigure_resource_for_mode(live_slack_client, hca_env),
+            "snapshot_config": project_snapshot_creation_config,
+            "dagit_config": preconfigure_resource_for_mode(dagit_config, hca_env),
+        },
+        config={
+            "resources": {
+                "snapshot_config": {
+                    "config": {
+                        "managed_access": False,
+                        "source_hca_project_id": "",
+                        "qualifier": None,
+                        "atlas": "hca"
+                    }
+                }
+            },
+        },
+        executor_def=in_process_executor
+    )
+
+
+@success_hook(
+    required_resource_keys={'slack', 'snapshot_config', 'dagit_config', 'hca_manage_config'}
+)
+def make_public_start_notification(context: HookContext) -> None:
+    context.log.info(f"Solid output = {context.solid_output_values}")
+    snapshot_id = context.solid_output_values["result"]
+    lines = ("HCA Starting Make Snapshot Public",
+             f"Snapshot name: {context.resources.snapshot_config.snapshot_name}",
+             f"Snapshot ID: {snapshot_id}",
+             f"Google Project ID: {context.resources.hca_manage_config.google_project_name}",
+             f"Dataset: {context.resources.snapshot_config.dataset_name}",
+             "Dagit link: " + f"<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>"
+             )
+
+    slack_msg_text = "\n".join(lines)
+    context.resources.slack.send_message(text=slack_msg_text)
+
+
+@success_hook(
+    required_resource_keys={'slack', 'snapshot_config', 'dagit_config', 'hca_manage_config', 'data_repo_client'}
+)
+def message_for_snapshot_public(context: HookContext) -> None:
+    snapshot_id = context.solid_output_values["result"]
+    snapshot_details = context.resources.data_repo_client.retrieve_snapshot(
+        id=snapshot_id,
+        include=["DATA_PROJECT"]
+    )
+    data_repo_project = snapshot_details.data_project
+
+    lines = ("HCA Snapshot Public",
+             f"Snapshot name: {context.resources.snapshot_config.snapshot_name}",
+             f"Dataset Google Project ID: {context.resources.hca_manage_config.google_project_name}",
+             f"Source Dataset: {context.resources.snapshot_config.dataset_name}",
+             f"Snapshot Google Data Project ID: {data_repo_project}",
+             "Dagit link: " + f"<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>"
+             )
+    slack_msg_text = "\n".join(lines)
+    context.resources.slack.send_message(text=slack_msg_text)
+
+
+@graph
+def set_snapshot_public() -> None:
```

Comment:
Can you add a basic unit test around this pipeline? I know it's kinda throwaway but it'd be good to have some coverage in the event this needs iterating.

Reply:
Oh you're right. I had that on my checklist but forgot about it when I started pushing to get this out to unblock the changed in the dedicatedIngestSA default value.
Looks like I'm going to have to cherry pick that now anyway, so I'll make sure to add this to my checklist